src/controller/cartController.js

```diff
@@ -45,21 +46,136 @@ exports.addToCart = async (req, res) => {
 exports.removeFromCart = async (req, res) => {
   try {
     // Delete data from cart using cart id
-    await cartModel.findOneAndDelete({ userId: req.params.userId });
-    res.status(200).send("data deleted sucessfully from cart");
+    await cartModel.findOneAndDelete({ userid: req.params.userid });
+    res.status(200).send("data deleted successfully from cart");
   } catch (err) {
-    res.status(400).send(err.message + "Data Deletion Unsucessufl from cart");
+    res.status(400).send(err.message + "Data Deletion Unsuccessful from cart");
   }
 };
 
 // View the user data from cart
 exports.findCart = async (req, res) => {
   try {
-    const cartData = await cartModel
-      .findOne({ userId: req.params.userId })
-      .lean();
+    const cartData = await cartModel.aggregate([
+      {
+        $match: {
+          userid: new mongoose.Types.ObjectId("65a76d78ca09b0052681193a"),
+        },
+      },
+      {
+        $lookup: {
+          from: "users",
+          localField: "userid",
+          foreignField: "_id",
+          as: "user",
+        },
+      },
+      {
+        $unwind: {
+          path: "$user",
+        },
+      },
+      {
+        $unwind: {
+          path: "$products",
+        },
+      },
+      {
+        $lookup: {
+          from: "products",
+          localField: "products.productid",
+          foreignField: "_id",
+          as: "product",
+        },
+      },
+      {
+        $unwind: {
+          path: "$product",
+        },
+      },
+      {
+        $group: {
+          _id: "$userid",
+          user: {
+            $first: {
+              _id: "$user._id",
+              name: "$user.name",
+              birthdate: "$user.birthdate",
+              email: "$user.email",
+              age: "$user.age",
+            },
+          },
+          products: {
+            $push: {
+              productid: "$products.productid",
+              quantity: "$products.quantity",
+              product: {
+                _id: "$product._id",
+                name: "$product.name",
+                description: "$product.description",
+                price: "$product.price",
+              },
+            },
+          },
+        },
+      },
```

Comment:
no need to add, you can comment it out for your reference

Reply:
resolved